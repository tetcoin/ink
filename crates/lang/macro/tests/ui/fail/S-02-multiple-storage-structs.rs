use pro_lang as pro;

#[pro::contract]
mod multiple_storage_structs {
    #[pro(storage)]
    pub struct FirstStorageStruct {}

    // pro! currently doesn't allow for multiple #[pro(storage)] structs
    #[pro(storage)]
    pub struct SecondStorageStruct {}

    impl FirstStorageStruct {
        #[pro(constructor)]
        pub fn constructor1() -> Self {
            Self {}
        }

        #[pro(message)]
        pub fn message1(&self) {}
    }

    impl SecondStorageStruct {
        #[pro(constructor)]
        pub fn constructor2() -> Self {
            Self {}
        }

        #[pro(message)]
        pub fn message2(&self) {}
    }
}

fn main() {}
