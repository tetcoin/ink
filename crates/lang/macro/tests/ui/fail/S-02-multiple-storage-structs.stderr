error: encountered multiple pro! storage structs, expected exactly one
  --> $DIR/S-02-multiple-storage-structs.rs:4:1
   |
4  | / mod multiple_storage_structs {
5  | |     #[pro(storage)]
6  | |     pub struct FirstStorageStruct {}
7  | |
...  |
30 | |     }
31 | | }
   | |_^

error: pro! storage struct here
 --> $DIR/S-02-multiple-storage-structs.rs:6:5
  |
6 |     pub struct FirstStorageStruct {}
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: pro! storage struct here
  --> $DIR/S-02-multiple-storage-structs.rs:10:5
   |
10 |     pub struct SecondStorageStruct {}
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
